libbeat/processors/actions/decode_json_fields_test.go

```diff
@@ -0,0 +1,100 @@
+package actions
+
+import (
+	"testing"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/stretchr/testify/assert"
+)
+
+var fields = [1]string{"msg"}
+var config, _ = common.NewConfigFrom(map[string]interface{}{
+	"fields": fields,
+})
+
+func TestMissingKey(t *testing.T) {
+	input := common.MapStr{
+		"pipeline": "us1",
+	}
+
+	actual := getActualValue(t, config, input)
+
+	expected := common.MapStr{
+		"pipeline": "us1",
+	}
+
+	assert.Equal(t, expected.String(), actual.String())
+}
+
+func TestFieldNotString(t *testing.T) {
+	input := common.MapStr{
+		"msg":      123,
+		"pipeline": "us1",
+	}
+
+	actual := getActualValue(t, config, input)
+
+	expected := common.MapStr{
+		"msg":      123,
+		"pipeline": "us1",
+	}
+
+	assert.Equal(t, expected.String(), actual.String())
+
+}
+
+func TestInvalidJSON(t *testing.T) {
+	input := common.MapStr{
+		"msg":      "{\"log\":\"{\\\"level\\\":\\\"info\\\"}\",\"stream\":\"stderr\",\"count\":3",
```

Comment:
If you create a raw string literal by surrounding the string with back ticks (`) then you don't need to use escaping on those quotes. I don't care if you fix this, it's more of a golang tip to keep you sane.